pkg/operator/apis/monitoring/v1/pod_types.go

```diff
@@ -245,17 +243,34 @@ type ScrapeEndpoint struct {
 // TargetLabels configures labels for the discovered Prometheus targets.
 type TargetLabels struct {
 	// Pod metadata labels that are set on all scraped targets.
-	// Permitted keys are `pod`, `container`, and `node` for PodMonitoring and
-	// `pod`, `container`, `node`, and `namespace` for ClusterPodMonitoring. The `container`
+	// Permitted keys are `container`, `node`, `pod`, `top_level_controller_name`,
+	// and `top_level_controller_type`. The `container`
+	// label is only populated if the scrape port is referenced by name.
+	// Defaults to [container, pod, top_level_controller_name, top_level_controller_type].
+	// If set to null, it will be interpreted as the empty list.
+	// This is for backwards-compatibility only.
+	// +kubebuilder:validation:items:Enum=container;node;pod;top_level_controller_name;top_level_controller_type
+	// +listType=set
+	// +kubebuilder:default=container;pod;top_level_controller_name;top_level_controller_type
+	Metadata *[]string `json:"metadata,omitempty"`
+	// Labels to transfer from the Kubernetes Pod to Prometheus target labels.
+	// Mappings are applied in order.
+	// +kubebuilder:validation:MaxItems=100
+	FromPod []LabelMapping `json:"fromPod,omitempty"`
+}
+
+// ClusterTargetLabels configures labels for the discovered Prometheus targets.
+type ClusterTargetLabels struct {
```

Comment:
question: is there any risk of backward compatibility by changing the go struct in the type here?

Reply:
SGTM